atomics/include/desul/atomics/Compare_Exchange_GCC.hpp

```diff
@@ -16,18 +16,15 @@ SPDX-License-Identifier: (BSD-3-Clause)
 //#define DESUL_HAVE_16BYTE_COMPARE_AND_SWAP
 #endif
 namespace desul {
-template <class MemoryOrderDesul>
-struct GCCMemoryOrder;
-
-template <>
-struct GCCMemoryOrder<MemoryOrderRelaxed> {
-  static constexpr int value = __ATOMIC_RELAXED;
-};
+template<class MemoryScope>
+void atomic_thread_fence(MemoryOrderAcquire, MemoryScope) {
+  __atomic_thread_fence(__ATOMIC_ACQUIRE);
+}
 
-template <>
-struct GCCMemoryOrder<MemoryOrderSeqCst> {
-  static constexpr int value = __ATOMIC_SEQ_CST;
-};
+template<class MemoryScope>
+void atomic_thread_fence(MemoryOrderRelease, MemoryScope) {
+  __atomic_thread_fence(__ATOMIC_RELEASE);
+}
```

Comment:
Better (?):

```
template <class MemoryOrder, class MemoryScope>
void atomic_thread_fence(MemoryOrder, MemorySpace) {
  __atomic_thread_fence(GCCMemoryOrder<MemoryOrder>::value);
}
```

Reply:
the main reason I didn't do it is so it doesn't conflict with the fully generic version, if there was one, but for atomic_thread_fence its fine I guess, since a generic version doesn't make sense. Either you hit the low level fence or you don't.